src/core/channels/presence_channel.ts

```diff
@@ -41,12 +42,28 @@ export default class PresenceChannel extends PrivateChannel {
     });
   }
 
-  /** Handles presence and subscription events. For internal use only.
+  /** Handles presence and subscription messages. For internal use only.
    *
-   * @param {String} event
-   * @param {*} data
+   * @param {Message} message
    */
-  handleEvent(event : string, data : any) {
+  handleMessage(message: Message) {
+    var event = message.event;
+    if (event.indexOf("pusher_internal:") === 0) {
+      this.handleInternalMessage(message)
+    } else {
+      var data = message.data;
+      if (message.user_id) {
+        var metadata = {};
+        metadata['user_id'] = message.user_id;
+        this.emit(event, data, metadata);
+      } else {
+        this.emit(event, data);
```

Comment:
I think that it's preferable to pass in an empty hash for the metadata in all cases, avoiding any possible errors from `undefined`.

Reply:
Agreed